# Day 3 EOD: The Banana Index

Date: 09/02/2026

df[df['column'] > value]                                  # the filter pattern


df.sort_values('column', ascending=False).head(n)         # the top-N pattern, whole table


df['column'].sort_values(ascending=False).head(n)         # the top-N pattern, one column

In [9]:
import pandas as pd
url = 'https://github.com/TheEconomist/banana-index-data/releases/download/1.0/bananaindex.csv'
df = pd.read_csv(url, index_col='entity')


In [13]:
df = df.drop(['year', 'Banana values', 'type', 'Chart?', 'Unnamed: 16'], axis='columns')

### Part 1: Get your bearings

In [16]:
df.shape

(160, 11)

#### 1. How many foods are in the dataset, and how many columns are left after the drop?

    - There are 160 foods in the dataset since that is what we indexed

In [17]:
df.head()

,emissions_kg,emissions_1000kcal,emissions_100g_protein,emissions_100g_fat,land_use_kg,land_use_1000kcal,Land use per 100 grams of protein,Land use per 100 grams of fat,Bananas index (kg),Bananas index (1000 kcalories),Bananas index (100g protein)
entity,,,,,,,,,,,
Ale,0.488690,0.317338,0.878525,2.424209,0.811485,0.601152,1.577687,3.065766,0.559558,0.362340,0.113771
Almond butter,0.387011,0.067265,0.207599,0.079103,7.683045,1.296870,3.608433,1.495297,0.443134,0.076804,0.026885
Almond milk,0.655888,2.222230,13.595512,4.057470,1.370106,2.675063,12.687839,4.600530,0.751002,2.537364,1.760651
Almonds,0.602368,0.105029,0.328335,0.119361,8.230927,1.423376,4.261040,1.610136,0.689721,0.119923,0.042520
Apple juice,0.458378,0.955184,29.152212,19.754980,0.660629,1.382839,43.232158,26.246743,0.524851,1.090638,3.775280


#### 2. Display the first few rows. What do the three columns whose names begin with Bananas index contain?

    - The three columns contain the kg, 1000 kcal, and 100g of protein.

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 160 entries, Ale to Yoghurt
Data columns (total 11 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   emissions_kg                       160 non-null    float64
 1   emissions_1000kcal                 160 non-null    float64
 2   emissions_100g_protein             158 non-null    float64
 3   emissions_100g_fat                 160 non-null    float64
 4   land_use_kg                        160 non-null    float64
 5   land_use_1000kcal                  160 non-null    float64
 6   Land use per 100 grams of protein  158 non-null    float64
 7   Land use per 100 grams of fat      160 non-null    float64
 8   Bananas index (kg)                 160 non-null    float64
 9   Bananas index (1000 kcalories)     160 non-null    float64
 10  Bananas index (100g protein)       160 non-null    float64
dtypes: float64(11)
memory usage: 19.1+ KB


#### 3. Run .info(). Two columns have missing values. Which, and how many?

    - The columns 'emissions_100g_protein' and 'land use per 100 grams of protein' are both missing two recordings. 

In [19]:
df.loc['Bananas', 'Bananas index (kg)']

1.0

#### 4. What is the value of Bananas index (kg) for Bananas itself? Explain in a sentence why the number has to come out that way.

    - The value is 1. The number came out that way because the index is emissions and when it is divided by the banana emisssion (1/1) it will equal 1.

### Part 2: Three Rankings